src/query/phrase_query/phrase_query.rs

```diff
@@ -56,9 +57,15 @@ impl PhraseQuery {
         PhraseQuery {
             field,
             phrase_terms: terms,
+            slop: 0,
         }
     }
 
+    /// Slop allowed for the phrase.
```

Comment:
Please document the behavior, with examples and an example where this is useful.

For instance "java developer"~1 matching "java software developer"

Reply:
You need to document the "slop keeps the order", and also the slop is "not an overall budget" specification.

In other words
"senior java developer"~1 does match "senior level distributed java software developer". 
The slop is not a budget that is spread amongst the different tokens. It is a maximum quantity that can be applied on every positions